predicate/predicate.py

```diff
@@ -357,3 +358,38 @@ def get_values_list(obj, *lookups, **kwargs):
     else:
         return [tuple(value_dict[lookup].value for lookup in lookups)
                 for value_dict in value_dicts]
+
+
+class PredicateQuerySet(object):
+    """
+    Iterable wrapper that follows the QuerySet API.
+    """
+    def __init__(self, iterable):
+        self.iterable = list(iterable)
+
+    def __repr__(self):
+        data = self.iterable[:REPR_OUTPUT_SIZE + 1]
+        if len(data) > REPR_OUTPUT_SIZE:
+            data[-1] = "...(remaining elements truncated)..."
+        return '<PredicateQuerySet %r>' % data
+
+    def __iter__(self):
+        return iter(self.iterable)
+
+    def all(self):
+        return self
+
+    def filter(self, **kwargs):
+        return self.__class__(P(**kwargs).filter(self.iterable))
+
+    def exclude(self, **kwargs):
+        return self.__class__(P(**kwargs).exclude(self.iterable))
+
+    def get(self, **kwargs):
+        return P(**kwargs).get(self.iterable)
+
+    def exists(self):
+        return bool(self.iterable)
+
+    def count(self):
+        return len(self.iterable)
```

Comment:
Need the following magic methods:
- `__getitem__`
- `__and__`
- `__or__`
- `__nonzero__`